packages/@cdktf/provider-generator/lib/get/generator/emitter/struct-emitter.ts

```diff
@@ -38,12 +54,14 @@ export class StructEmitter {
         this.code.line(
           `* Docs at Terraform Registry: {@link ${resource.linkToDocs}#${att.terraformName} ${resource.className}#${att.terraformName}}`
         );
+        this.warnAboutIdField(att);
```

Comment:
Is there a way we can figure out which provider sdk version has been used? Probably not, right?

Reply:
Unfortunately this information is not part of the schema, I think one would need to monkey type it, e.g. if there is a nested type present somewhere it's a newer one. Not sure if it's worth the effort though

Reply:
Ah, yeah, not worth it then 😇